### ISSUE TITLE
Describe Relationship to JWKS

### ISSUE THREAD

**Issue by OR13** [w3c-ccg/webkms#2]
web crypto makes heavy use of JOSE, we should clearly define how WebKMS works with JWKS, particularly as it relates to overlapping functionality of storing extractable keys and meta data.

Is JWKS a viable serialization of a WebKMS datastore (assuming 100% extractable keys).

**Comment by OR13** [comment]
We have a library https://github.com/transmute-industries/did-wallet which we use to store key material and secrets such as mnemonics... We'd love to have a more standards based approach to the problem of serialized wallet files

**Comment by OR13** [comment]
https://github.com/hyperledger/indy-sdk/tree/master/docs/design/009-wallet-export-import 

Also has a JSON Serialization format, would be nice to figure out how to go from structured JSON to initialized WebKMS Switch / Driver.

**Comment by ChristopherA** [comment]
My problem with many of the KM architectures that I'm seeing here is that they are single-key centric, whereas almost everything I do starts master SECRET management, not KEY management, and the master secret can be used to derive hierarchical keys (and other keys). Note that this master secret is public key cryptography agnostic — it can be used to create RSA keys, ECDSA keys, 25519 keys, etc.

We basically start with a 512 bit entropy secret, which is expressed in a binary form with optional metadata (for instance the key birthday to speed up blockchain lookups) for online use, or offline in 12 or 24 word mnemonic seed (using BIP39 or my iambic pentameter memorization approach), or a series of 20 or 40 word SLIP39 shares. I have a book out on best practices to store this secret offline https://www.smartcustody.com/2019-09-17-Blockchain-Commons-Releases-First-SmartCustody-Book/

See also this demo where we use the iOS native image recognization AI to read the mnemonic words directly into an offline iOS app https://twitter.com/FullyNoded/status/1214545661233745921?s=20 where they can be used for computation but are never stored.

-- Christopher Allen

**Comment by OR13** [comment]
Yes, we also use BIP39... Its fairly common to export and import keys using it in the blockchain world. It would be awesome to account for that in a standards compliant keystore file. In our help libraries today, we store the phrase encrypted along side private keys.

I would love for a single keystore format, encrypted at rest, annotated with an extensible type system, that we could use a the import source for extractable KMS systems, like web crypto.

When storing meta data about derived keys in a "wallet/keystore" file, you need to include the derivation path along with the metadata... this helps you remember things like:

I used this derived key in this context, its a correlation risk... etc...

At pretty much every IIW I have attended this gets discussed, and everyone agrees that users who don't annotate their keys, can't maintain anonymity, and yet there is no open standards compliant key annotation system...

This is one of the major concerns I have with WebKMS, if I can't export my keys with metadata about their use, I'm very likely to export and import and wreck myself from a privacy perspective.

Thats why I would love to figure out a simple serialization format that a user could look at in plain text, and see things like: "I used this key1 with did1, mnemonic2 with did2, etc..."
